### PR TITLE
feat: Added family details get endpoint #46

### PR DESCRIPTION
**Pull request by Thealean28**
## Description
Now it is possible to get the details of every family stored in the database.

## Related Issue(s)
- #46 

## Checklist
- [x] I have read and followed the project's contributing guidelines.
- [x] I have tested the changes locally.
- [x] I have made corresponding changes to the documentation if necessary.
- [x] My code follows the project's code style guidelines.

## Reviewers
- @alvgonfri 
- @josgarber6 

## Tester
- @davidg43

### CODE REVIEW

**Review by josgarber6 — DISMISSED** [review]
Everything works correctly, well done

**Comment by davidg43** [comment]
Now it is possible to test the api call for family details! Due to an unexpected failure I cant run the tests so I request a revision to run them as soon as possible.

**Review by alvgonfri — CHANGES_REQUESTED** [review]
- The URL must be changed to `/api/v1/cyc/family/{family_id}`. It's important to change it in the test, too:
image

- The test is not working. Please take the test uploaded in this commit as an example of creating a family: https://github.com/ISPP-07/backend/pull/49/commits/f43f184fccb4bc98b5d26aaacfb2197c99681694. @davidg43

File: tests/api/test_cyc_family_details.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            person1_data,
+            person2_data
+        ]
+    }
+    family = await Family.create(session, **family_data)
+    return family
+
+
+def test_get_family_details(client: TestClient, create_family: Family):
+    family_id = create_family["id"]
```
Comment (alvgonfri) [inline tests/api/test_cyc_family_details.py:44]: Change to `create_family.id`

File: tests/api/test_cyc_family_details.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import pytest
+from fastapi.testclient import TestClient
+
+from src.core.config import settings
+
+from src.modules.cyc.family.model import Family
+
+
+@pytest.mark.asyncio
```
Comment (alvgonfri) [inline tests/api/test_cyc_family_details.py:9]: Change to `@pytest_asyncio.fixture`

File: tests/api/test_cyc_family_details.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import pytest
```
Comment (alvgonfri) [inline tests/api/test_cyc_family_details.py:1]: Change to `pytest_asyncio`

File: tests/api/test_cyc_family_details.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def test_get_family_details(client: TestClient, create_family: Family):
+    family_id = create_family["id"]
+    url = (f'{settings.API_STR}'
+           f'cyc/family/details/' + str(family_id))
+
+    response = client.get(url)
+    assert response.status_code == 200
+    assert response.json() == create_family.model_dump()
```
Comment (alvgonfri) [inline tests/api/test_cyc_family_details.py:50]: I think this is not going to work. Maybe you could compare `response.json()` to the json you expect to be returned.

File: tests/api/test_cyc_family_details.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "id": 1,
+        "name": "La Familia",
+        "phone": "123456789",
+        "address": "123 Main St",
+        "number_of_people": 4,
+        "referred_organization": "Hospital ABC",
+        "next_renewal_date": "2100-12-31",
+        "derecognition_state": "Active",
+        "obsarvations": "Observation 1",
+        "members": [
```
Comment (alvgonfri) [inline tests/api/test_cyc_family_details.py:34]: This is not how you create a person. You must use the `Person.create` method as you did when creating a Family, indicating the "family_id". The "members" attribute must disappear from `family_data`.

**Review by alvgonfri — CHANGES_REQUESTED** [review]
It's better now, but there are still some problems in the tests. @davidg43 

Also, the URL must be changed as indicated in my previous comment.

File: tests/api/test_cyc_family_details.py (hunk 5 above)
Comment (alvgonfri) [reply tests/api/test_cyc_family_details.py:34]: These are the lines I'm talking about:
image

**Review by jmartinacu — APPROVED** [review]
Ok

**Review by alvarobernal2412 — APPROVED** [review]
Perfect!

**Pull request merged**